app/src/views/search-engine/index.js

```diff
@@ -0,0 +1,31 @@
+import React, { useState, useEffect } from 'react';
+import { useQuery } from '@apollo/client';
+import { useDebounce } from 'use-lodash-debounce';
+import { Container } from '@material-ui/core';
+import SearchBar from './components/search-bar';
+import ArticlesList from './components/articles-list';
+import { GET_ARTICLES } from './graphql/queries';
+
+const SearchEngine = () => {
+  const [articles, setArticles] = useState([]);
+  const [search, setSearch] = useState('');
+  const debouncedSearch = useDebounce(search, 1000);
+  const getArticlesQuery = useQuery(GET_ARTICLES, { variables: { search: debouncedSearch } });
+
+  useEffect(() => {
+    const { data } = getArticlesQuery;
+
+    if (data?.articles) {
```

Comment:
Can you please explain what this operation does. If it's something not that known a single short comment would be nice. If it's something common then just answer my comment and leave it as is

Reply:
These are known as [React Hooks](https://reactjs.org/docs/hooks-state.html#whats-a-hook). It was a feature added on recent iterations of React. It's basically the newest way to handle state within your components in an event-driven way.

This Hook is just querying the Apollo Server to fetch the articles and storing them in local state to display them.